virtual_ship/virtual_ship_configuration.py

```diff
@@ -1,173 +1,124 @@
+
+    fieldset: FieldSet
+    max_depth: float
+    drift_depth: float
+    vertical_speed: float
+    cycle_days: float
+    drift_days: float
+
+
+@dataclass
+class ADCPConfig:
+    """Configuration for ADCP instrument."""
+
+    max_depth: float
+    bin_size_m: int
+
+
+@dataclass
+class VirtualShipConfig:
+    """Configuration of the virtual ship."""
+
+    start_time: datetime
+    route_coordinates: list[Location]
 
     adcp_fieldset: FieldSet
     ship_underwater_st_fieldset: FieldSet
-    argo_float_fieldset: FieldSet
-    drifter_fieldset: FieldSet
     ctd_fieldset: FieldSet
     drifter_fieldset: FieldSet
-    argo_float_fieldset: FieldSet
-
-    def __init__(
-        self,
-        json_file,
-        adcp_fieldset: FieldSet,
-        ship_underwater_st_fieldset: FieldSet,
-        ctd_fieldset: FieldSet,
-        drifter_fieldset: FieldSet,
-        argo_float_fieldset: FieldSet,
-    ):
+
+    argo_float_deploy_locations: list[Location]
+    drifter_deploy_locations: list[Location]
+    ctd_deploy_locations: list[Location]
+
+    argo_float_config: ArgoFloatConfig
+    adcp_config: ADCPConfig
+
+    def verify(self) -> None:
         """
-        Initialize this object.
-
-        :param json_file: Path to the JSON file to init from.
-        :param adcp_fieldset: Fieldset for ADCP measurements.
-        :param ship_underwater_st_fieldset: Fieldset for ship salinity temperature measurements.
-        :param ctd_fieldset: Fieldset for CTD measurements.
-        :param drifter_fieldset: Fieldset for CTD measurements.
-        :param argo_float_fieldset: Fieldset for argo float measurements.
-        :raises ValueError: If JSON file not valid.
+        Verify this configuration is valid.
+
+        :raises ValueError: If not valid.
         """
-        self.adcp_fieldset = adcp_fieldset
-        self.ship_underwater_st_fieldset = ship_underwater_st_fieldset
-        self.ctd_fieldset = ctd_fieldset
-        self.drifter_fieldset = drifter_fieldset
-        self.argo_float_fieldset = argo_float_fieldset
-
-        with open(json_file, "r") as file:
-            json_input = json.loads(file.read())
-            for key in json_input:
-                setattr(self, key, json_input[key])
-
-        # Create a polygon from the region of interest to check coordinates
-        north = self.region_of_interest["North"]
-        east = self.region_of_interest["East"]
-        south = self.region_of_interest["South"]
-        west = self.region_of_interest["West"]
-        poly = Polygon([(west, north), (west, south), (east, south), (east, north)])
-        # validate input, raise ValueErrors if invalid
-        if (
-            self.region_of_interest["North"] > 90
-            or self.region_of_interest["South"] < -90
-            or self.region_of_interest["East"] > 180
-            or self.region_of_interest["West"] < -180
-        ):
-            raise ValueError("Invalid coordinates in region of interest")
```

Comment:
Well, I think the test that latitude is between -90 and 90 is stil valid, so I'd keep that in. And we can test that longitude is either between -180 and 180, or between 0 and 360. In my last comment, I didn't mean to propose to complete remove these checks

Reply:
I am not sure if that makes sense, because we cannot deduce what range the user wants to use. Functionally it does not matter, and such checks might only annoy the user? I guess we could check for larger than -180 and smaller than 360, but I'm not sure if that is useful. It's a bit odd if a user provides 720, but it's technically fine?